### ISSUE TITLE
redesign of the forum in general

### ISSUE THREAD

**Issue by surmvoise** [processing/processing-forum#17]
I guess I'm not the only one with this opinion but I couldn't find any posts about my concerns on the web so far. So here's mine.
I think for such a great community like the Processing community there's a much better forum needed. I find it terribly hard to search for specific things, even through categories. (How do 20.1K categories help?) And also the search results are pretty bad. Every post with a matched keyword is listed to it's full extent and searching for the corresponding "tags" isn't even suggested.
Basically what I would love to see is an overall redesign of the forum. Maybe it's already in the works, then i'd be happy to hear it. :)

**Comment by alignedleft** [comment]
A redesign is not in the works. This is an all-volunteer effort — installing, maintaining, designing. Would you like to volunteer? :-)

**Comment by surmvoise** [comment]
Oh, what a pity. I'm unfortunately not really a programmer – besides that little bit of Processing I do. Though I'm a graphic designer. If you feel like you could need some assistance in that field I'd be happy to help.

**Comment by alignedleft** [comment]
I believe @fjenett is still in charge of the forums' back-end here. It seems every few years there is a push to overhaul them, as our needs evolve (perhaps rightfully so).

**Comment by hamoid** [comment]
What are your thoughts on using https://www.discourse.org/ ? I think that forum is a pleasure to use.

**Comment by REAS** [comment]
Hello @hamoid. Yes, I think somethings needs to be done. When we last switched over the forum, Discourse was not ready, but it looked promising. I'm going to make a post on the Forum to ask the question.

**Comment by surmvoise** [comment]
Yes, Discourse seems like a brilliant solution to me. 
Great, @REAS. Please also leave a link to your forum post here so we can join the discussion.

**Comment by hamoid** [comment]
In case this is eventually attempted, I found here a script that could be used to import Vanilla into Discourse: https://meta.discourse.org/t/how-to-migrate-from-vanilla-to-discourse/27273

**Comment by REAS** [comment]
Thank you for finding that! We've re-booted the Forum four times over the years. I think we should discuss if importing is a good plan or if we should start fresh. I'm open to either. For me, the largest questions are social, meaning the way a new forum would be moderated.

**Comment by EmmanuelPil** [comment]
Before starting a new forum I would gather all that is wrong with this one, to see if the new one solves it. Like for instance when you copy a long code example, and paste it, you get an extra new line between  the existing.  A work arround is to past it first in for instance, a new github text input box, and then copy it again. The code will be ' cleaned'. Another issue is when you use <> in the code like List< File >alfiles = Arrays.asList(files);  you have to put spaces before and behind 'File' otherwise the word File will just disappear. Another thing is when you write @Overide  a lot of html code will be added. The forum also isn't  android friendly; the main block is far to narrow and you have to zoom in and out all the time, further you have the search problems already mensioned. I know I had a lot more issues, but this is what came to my mind at this moment.

**Comment by hamoid** [comment]
@EmmanuelPil There was a longer discussion about this topic in the forum: https://forum.processing.org/two/discussion/26613/is-it-time-for-a-new-forum#latest Also, for a few weeks there was a test instance of Discourse where we did tests and wrote feedback.

Good points, the copy-pasting issues should be tested as well. Maybe you can create an account at https://try.discourse.org/ and see if those issues exist there? Cheers!

**Comment by fjenett** [comment]
New forum here: https://discourse.processing.org/

**Comment by surmvoise** [comment]
Awesome!
But the Forum link on processing.org still directs to https://forum.processing.org/two/

**Comment by REAS** [comment]
Yes, thank you. We're "soft launching" the forum and we will have both open for awhile. We'll move the link over soon, when we feel the new forum is ready for more traffic.
